### ISSUE TITLE
Asustor 604T no bios or usb boot

### ISSUE THREAD

**Issue by Ogglord** [mafredri/asustor-platform-driver#5]
Hi Mafredi,

I've got an otherwise fully functional asustor 604t, I set out to install debian on it. But I can't enter bios or get it to boot from my USB, I've tried smashing ESC, F2, DEL during boot. But I get no HDMI output, i've tried numerous hdmi cables and two monitors. It just boots straight to ADM. I have removed my 4 spinning rusts and I'm only using an 2.5inch SSD at the moment, with a fresh ADM OS on it.

Any suggestions, am I missing something obvious

**Comment by mafredri** [comment]
Hi @Ogglord, I feel your pain, unfortunately the AS-604T doesn't have a proper BIOS, it's basically a very dirty hacked-together EFI bootloader that has everything hard-coded. 😅 It's also impossible to get video out before the appropriate kernel module is loaded so you're limited to either doing things blind or hooking up to the serial port.

Have you seen my other repo https://github.com/mafredri/asustor_as-6xxt, and more specifically, https://github.com/mafredri/asustor_as-6xxt/blob/4bf6736ceb4cbe1675249cbefa91c570f289e12e/debian.md? Steps are quite abridged, but should give you a general idea of what to do. As long as you automate the install/live image bootup, and place your grub .efi in the correct location, it should work out just fine.

One step that might not be mentioned in that repo, and that you _may_ need to do, is hold the [USB COPY]-button (front panel, bottom left) whilst booting. This should force the AS-604T to load `asloader.efi` from the USB instead of the internal eMMC.

In case you have further questions, we can move this discussion over to https://github.com/mafredri/asustor_as-6xxt (feel free to open a new discussion there).

Hoppas det hjälper, och lycka till!

**Comment by Ogglord** [comment]
Okay, that explains a lot. I was quite close to pulling the trigger.

The thing holding me back is that once I disable the `asloader.efi` on the built-in flash, there is no turning back. If I fail the first boot from USB, what are the options then? Do I just keep tweaking the USB file until it boots and I get ssh access? Also I was not sure where to place the `preseed.cfg`. Some guides online extracted the `initrd.gz` and put the preseed in there, and compressed it again. Other guides just seem to put it in the root of the usb. How did you go about that?

 I have a serial to usb connector lying around somewhere (USB to DB9F serial cable), but that won't fit. 

Many thanks for helping out. This is just a quest for me to get rid of the ADM OS and run btrfs. :)

**Comment by mafredri** [comment]
> The thing holding me back is that once I disable the `asloader.efi` on the built-in flash, there is no turning back.

Yeah, that can be scary. That's why I mentioned the [USB Copy]-button work-around for forcing `asloader.efi` to be loaded from USB (USB drive needs to be put in a BACK-side USB port, though, front wont work). This way there's no need to disable the internal `asloader.efi`.

If you look at the `debian.md` document I shared, you can see examples of `grub.cfg` that use the preseed. Obviously there's no telling if my old preseed will work with the version you're using now, though.

**Comment by Ogglord** [comment]
Yes, but you had `/cdrom/preeseed.cfg` in the `grub.cfg` and then a comment saying `/cdrom/preeseed.cfg` = `(usb)/preeseed.cfg` that I don't fully understand. Is that a change I should do to my config (in verbatim) or is `/cdrom/` a special path that actually points to the USB? :) Cheers

**Comment by mafredri** [comment]
@Ogglord I don't remember the context anymore, but I do believe it's the latter, place it in the root of your USB (and keep `/cdrom` in the grub config) 👍🏻
